Public/Src/FrontEnd/Nuget/NugetResolver.cs

```diff
@@ -62,12 +69,89 @@ public override async Task<bool> InitResolverAsync(IResolverSettings resolverSet
+                      Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid )
+                {
+                    // Validation of existing cgmainfest.json results in failure due to mismatch. Should fail the build in this case.
+                    try
+                    {
+                        existingCgManifest = File.ReadAllText(Configuration.FrontEnd.ValidateCgManifestForNugets.ToString(Context.PathTable));
+                        FrontEndHost.Engine.RecordFrontEndFile(
+                            Configuration.FrontEnd.ValidateCgManifestForNugets,
+                            CGManifestResolverName);
+                    }
+                    // CgManifest FileNotFound, log error and fail build
+                    catch (DirectoryNotFoundException e)
+                    {
+                        Logger.ReportComponentGovernanceValidationError(Context.LoggingContext, "Cannot read Component Governance Manifest file from disk\n" + e.ToString());
+                        return false;
+                    }
+                    catch (FileNotFoundException e)
+                    {
+                        Logger.ReportComponentGovernanceValidationError(Context.LoggingContext, "Cannot read Component Governance Manifest file from disk\n" + e.ToString());
+                        return false;
+                    }
+                    if (!cgManfiestGenerator.CompareForEquality(generatedCgManifest, existingCgManifest))
+                    {
+                        Logger.ReportComponentGovernanceValidationError(Context.LoggingContext, @"Existing Component Governance Manifest file is outdated, please generate a new one using the argument /generateCgManifestForNugets:<path>");
+                        return false;
+                    }
+
+                    m_resolverState = State.ResolverInitialized;
+                    return true;
+                }
+
+                // GenerateCgManifestForNugets writes a new file when the old file does not match, hence it will always be valid and does not need validation
+
+                try
+                {
+                    // We are calling FrontEndHost.Engine.RecordFrontEndFile towards the end of this function because we may update this file after the read below
+                    // Updating the file will cause a hash mismatch and the build to fail if this file is read again downstream
+                    existingCgManifest = File.ReadAllText(Configuration.FrontEnd.GenerateCgManifestForNugets.ToString(Context.PathTable));
+                }
+                // CgManifest FileNotFound, continue to write the new file
+                // No operations required as the empty existingCgManifest will not match with the newly generated cgManifest
+                catch (DirectoryNotFoundException) { }
+                catch (FileNotFoundException) { }
+
+                if (!cgManfiestGenerator.CompareForEquality(generatedCgManifest, existingCgManifest))

```

Comment:
Why do you want to perform this check here?  Can't you just write the file to disk regardless if it already exists and whether its existing content is the same?